### ISSUE TITLE
Fix `hue-degree-notation` false negatives and positives for relative colors

### ISSUE THREAD

**Issue by jamesnw** [stylelint/stylelint#8983]
### What minimal example or steps are needed to reproduce the bug?

```css
a{
  color: oklch(from red 1 c h)
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  rules: {'hue-degree-notation': 'angle'}
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcICGwA6A7ABJyAbCATvJhANa5gAWAFAGYEQC2mBMUmAjDppQJQYBfEABoQtAJa4YAOSSM4iGAA85ABykA6MAGdtI8BHQSA5ghDLVhAC6ZYtJAFdcNtFhZOY24sADklBzAAtLDGrEHoEFZIVuKGPsQ+SOjGUj4CggDcGPqQRuLGAGKEjNFm2lYAnlK44uhWWoYmGowAVnqisKp6iK7YqCDlVTA1df3E-bjRnlb9whh9A5XVtVaBuSaB5UlQSARQY5gTU+X9gtkCQA

### Which Stylelint-related dependencies are you using?

latest- 
[stylelint](https://stylelint.io/)@17.0.0
[stylelint-config-standard](https://www.npmjs.com/package/stylelint-config-standard)@40.0.0

### What did you expect to happen?

No error

### What actually happened?

The following error was reported:

```
Expected "1" to be "1deg" (hue-degree-notation)
```

In the example, 1 is a lightness value, which is not a hue value, and degrees are not valid. However, it's the third value, which in non-relative syntax is hue, which does use degrees.

### Do you have a proposal to fix the bug?

Relative color syntax allows you to create a new color based on a given color, for instance `oklch(from red l c h)`. This shifts the channel values from 1,2,3 to 3,4,5.

Channel values should be shifted to accommodate relative color syntax when the `from` keyword is encountered.

**Comment by jeddy3** [comment]
@jamesnw Thanks for the report, demo and for using the template.

I've titled it as a false negative and positive bug.

False negative - the `0` in the following [isn't flagged](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvChANZ5gAWAFAGaEQC2KhMUKeuKADAJRYAviAA0IOgEs8MAHJImcRDAAe8gA7SAdGADOO0eAiZJAcwQgVaogBcUsOkgCueWxmytnMHSWAByKo4wALSwJmzBmBDWSNYSRr4kvkiYJtK+gkIA3FgGkMYSJgBiREwx5jrWAJ7SeBKY1tpGpppMAFb6YrBq+ohuOOggFdUwtfUDJAN4MV7WAyJY-YNVNXXWQXmmQRXJUEiEUOMok9MVA0I5gkA):

```css
a {
  color: oklch(from red l c 0)
}
```

False positive - the `0` in the following [is flagged](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdilkA2EATvChANZ5gAWAFAGaEQC2KhMUKADLilQJRYAviAA0IOgEs8MAHJImcRDAAe8gA7SAdGADOO0eAiZJAcwQgVaogBcUsOkgCueWxmytnMHSWAByKo4wALSwJmzBmBDWSNYSRr4kvkiYJtK+gkIA3FgGkMYSJgBiREwx5jrWAJ7SeBKY1tpGpppMAFb6YrBq+ohuOOggFdUwtfUDJAN4MV7WAyJY-YNVNXXWQXmmQRXJUEiEUOMok9MVA0I5gkA):

```css
a {
  color: oklch(from red 0 c h)
}
```

> Channel values should be shifted to accommodate relative color syntax when the `from` keyword is encountered.

SGTM. [Spec for reference](https://drafts.csswg.org/css-color-5/#relative-colors) for those interested.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by ragini-pandey** [comment]
Hello Team,

I’ve raised a PR for this issue => https://github.com/stylelint/stylelint/pull/8990.

If it’s okay, could you please assign this issue to me?

I’d be happy to continue working on it and address any feedback.
